gpjax/utils.py

```diff
@@ -89,7 +88,7 @@ def dict_array_coercion(params: Dict) -> Tuple[Callable, Callable]:
     """
     flattened_pytree = jax.tree_util.tree_flatten(params)
 
-    def dict_to_array(parameter_dict) -> jnp.DeviceArray:
+    def dict_to_array(parameter_dict) -> jax.Array:
         return jax.tree_util.tree_flatten(parameter_dict)[0]
 
```

Comment:
Similar comment to `kernels.py` typing. Unless we can't determine the shape in this case?

Reply:
This is tricky as the shapes are dynaminc. Typing this really wouldn't add any "useful" information in my opinion. I'd suggest we leave it as an `Array` here.